### PR TITLE
kind: support config files

### PR DESCRIPTION
**Pull request by BenTheElder**
- refactor config structs in cluster package, move the name to context and use config just for create
- add functionality to load create config from file
- add flag for config file to create cli
- add kubeadm config template to create command

/area kind

### CODE REVIEW

**Comment by BenTheElder** [comment]
er need to refactor the config tests... something got lost in rebase heck there. will fix
/hold

**Comment by BenTheElder** [comment]
/hold cancel
fixed the tests

File: kind/pkg/cluster/cluster.go [hunk 1]
```diff
@@ -30,29 +32,72 @@ import (
...
+	Name string
 }
 
-// NewContext returns a new cluster management context with Config config
-func NewContext(config Config) *Context {
-	return &Context{
-		config: config,
+// similar to valid docker container names, but since we will prefix
+// and suffix this name, we can relax it a little
+// see Validate() for usage
```
Comment (krzyzacy) [inline kind/pkg/cluster/cluster.go]: ^^ source of the regex?

Comment (BenTheElder) [reply kind/pkg/cluster/cluster.go]: done

**Pull request merged**